heal/vlmd/validate/csv_validator.py

```diff
@@ -25,19 +28,65 @@ def vlmd_validate_csv(input_file: str, schema_type: str) -> bool:
     if schema_type not in ALLOWED_SCHEMA_TYPES:
         raise ValueError(f"Schema type must be in {ALLOWED_SCHEMA_TYPES}")
 
-    schema = get_schema(input_file, schema_type)
+    if isinstance(data_or_path, list):
+        logger.debug("Getting csv as json array from input object")
+        data = data_or_path
+        from_file = False
+    else:
+        logger.debug("Getting csv data from file")
+        data = read_delim(data_or_path).to_dict(orient="records")
+        from_file = True
+
+    if len(data) == 0:
+        raise ValueError(f"Could not read csv data from input")
+
+    schema = get_schema(data_or_path, schema_type)
     if schema == None:
         raise ValueError(f"Could not get schema for type = {schema_type}")
 
     schema = add_types_to_props(schema)
     logger.debug("Checking schema")
     jsonschema.validators.Draft7Validator.check_schema(schema)
 
-    data = read_delim(input_file).to_dict(orient="records")
-    if len(data) == 0:
-        raise ValueError(f"Could not read data from file {input_file}")
-
     logger.debug("Validating data")
-    jsonschema.validate(instance=data, schema=schema)
+    logger.debug(data)
+    # jsonschema.validate(instance=data, schema=schema)
```

Comment:
no longer needed?

Reply:
Yes, correct, that step is no longer needed. For CSV files, we first try the extraction. If that is successful the `validate`  module then tries the `jsonschema.validate` on the converted CSV data. (see the updated figure in the ticket [HP-1827](https://ctds-planx.atlassian.net/browse/HP-1827)). Most of the CSV checking is handled by `extract` and the `csv_validator` is now removed.